### PR TITLE
Ensure that router logs system health errors

### PR DESCRIPTION
**Pull request by pnarayanan**

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/router/RouterException.java [hunk 1]
```diff
@@ -21,12 +21,12 @@
   private final RouterErrorCode errorCode;
 
   public RouterException(String message, RouterErrorCode errorCode) {
-    super(message);
+    super(message + errorCode);
```
Comment (cgtz) [inline ambry-api/src/main/java/com.github.ambry/router/RouterException.java]: Minor: add a space in between the message and error code

**Comment by pnarayanan** [comment]
@cgtz addressed comment.

**Comment by cgtz** [comment]
There seems to be a test that checks an exception message:
```
com.github.ambry.frontend.AmbryBlobStorageServiceTest > routerExceptionPipelineTest FAILED
    org.junit.ComparisonFailure: Unexpected exception message expected:<VMTTRLPYD9[]> but was:<VMTTRLPYD9[ Error: UnexpectedInternalError]>
        at org.junit.Assert.assertEquals(Assert.java:123)
        at com.github.ambry.frontend.AmbryBlobStorageServiceTest.checkRouterExceptionPipeline(AmbryBlobStorageServiceTest.java:1545)
        at com.github.ambry.frontend.AmbryBlobStorageServiceTest.doRouterExceptionPipelineTest(AmbryBlobStorageServiceTest.java:1522)
        at com.github.ambry.frontend.AmbryBlobStorageServiceTest.routerExceptionPipelineTest(AmbryBlobStorageServiceTest.java:676)
```

**Comment by vgkholla** [comment]
Yeah because I think it sets the exception message - With this change, something extra is being added to the message.

**Comment by pnarayanan** [comment]
@cgtz fixed the test, this can be merged if you don't have any other comments.

**Pull request merged**